### DISCUSSION TITLE
Starring events takes takes time and exceptional computation

### DISCUSSION THREAD

**Discussion by thinkloop** [blakeblackshear/frigate#7728] [General]
Starring an event takes 5-10 seconds to complete. The python3 process spikes up to 300% meanwhile. Starring multiple events in a row can cause the system to become unresponsive requiring a reboot. 

```
Frigate: 0.12.0-DA3E197
```

```
Debian GNU/Linux trixie/sid
Codename:       trixie
```

**Comment by NickM-27** [comment]
This is going to be due to DB issues, recommend running `VACUUM;` on the DB.

The upcoming 0.13 also contains a number of DB fixes / optimizations including running VACUUM automatically

**Reply by thinkloop** [reply]
Thank you. How do I run VACUUM exactly?

**Reply by NickM-27** [reply]
https://github.com/blakeblackshear/frigate/issues/7565#issuecomment-1691647117

**Reply by thinkloop** [reply]
As an aside, it seems strange that the resolution is to VACUUM on such a small DB - are you sure it's not an underlying index issue?

**Reply by NickM-27** [reply]
Like I said previously, we already have many DB fixes / optimizations coming in 0.13. VACUUM is just another one of those that has shown a massive improvement for some users (and marginal for others), especially for users that have databases that are 1 year+ old because the data is so fragmented.

**Reply by NickM-27** [reply]
VACCUM is the easiest thing to recommend because 0.13 is not in beta yet
